ONSdigital/eq-author-app
# EAR 1096 - Refactor AJV schema + upgrade to v8

### What is the context of this PR?

[Ticket](https://collaborate2.ons.gov.uk/jira/browse/EAR-1096)

A general freshening up of our AJV validation schemas / keywords:

- Upgrades AJV from v6 -> v8.
- Review existing custom keywords, delete those replaceable with standard JSON schemas 
- Updates custom keywords to be compatible with AJV v8's interface
- Enables strict mode in v8 + fixes ensuing errors with existing schemas

As part of this, `maxLength` as a validation property on `TextArea` has been changed to be numerical rather than a string so that we can use standard AJV validation on it directly; UI code updated to save it as a number and not a string in saving mutation.

### How to review

- NB: You'll need to `yarn` in eq-author-api to upgrade AJV
- Check validation in general works as before
- Check TextArea max length user validations still work

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
